src/test/scala/com/ambiata/poacher/hdfs/ConfigurationTemporary.scala

```diff
@@ -0,0 +1,36 @@
+package com.ambiata.poacher.hdfs
+
+import com.ambiata.mundane.control._
+import com.ambiata.mundane.io._
+import org.apache.hadoop.conf.Configuration
+import com.nicta.scoobi.Scoobi._
+import scalaz._, Scalaz._, effect.IO
+
+case class ConfigurationTemporary(path: String, configuration: Configuration) {
+  override def toString: String =
+    s"ConfigurationTemporary($path)"
+
+  def conf: RIO[Configuration] = for {
+    d <- LocalTemporary(path).directory
+    _ = configuration.set("hadoop.tmp.dir", d.path)
+    _ = configuration.set("scoobi.dir", d.path + "/")
+  } yield configuration
+}
+
+object ConfigurationTemporary {
+  def randomPath(conf: Configuration): ConfigurationTemporary =
```

Comment:
This seems a little awkward. At least in this PR both callsites just pass in an empty `Configuration`. And then above we're mutating the same configuration on `conf` (which then returns it). I guess I would be tempted to have the one function that creates a `Configuration`, and sets the temporary paths. (And then you can set whatever you like on it) Thoughts?